stdlib/public/Concurrency/TaskGroup.cpp

```diff
@@ -1182,9 +1190,9 @@ void AccumulatingTaskGroup::offer(AsyncTask *completedTask, AsyncContext *contex
 
   // ==== a) has waiting task, so let us complete it right away
   if (assumed.hasWaitingTask()) {
-    resumeWaitingTask(completedTask, assumed, hadErrorResult);
-    unlock(); // TODO: remove fragment lock, and use status for synchronization
-    return;
+    // Must unlock before we resume the waiting task
+    unlock();
+    return resumeWaitingTask(completedTask, assumed, hadErrorResult);
```

Comment:
Same issue as https://github.com/apple/swift/pull/67700 fixed, but inside the offer impl